web/src/views/CreateDandisetView/CreateDandisetView.vue

```diff
@@ -159,7 +158,14 @@ if (!loggedIn()) {
 }
 
 async function registerDandiset() {
-  const metadata = { name: name.value, description: description.value, license: license.value };
+  const metadata: Record<string, any> = {
```

Comment:
Shouldn't this be `Record<"name" | "description", string>`?

Reply:
I'm guessing the reason this was done was to prevent the type checker from complaining when the `license` field is added below.

Reply:
Can you do
```
const metadata: {name: string, description: string, license?: LicenseType} = ...
```
:man_shrugging:

Reply:
> Can you do
> 
> ```
> const metadata: {name: string, description: string, license?: LicenseType} = ...
> ```
> 
> man_shrugging


Yeah, that syntax is much better. And my use of `any` here also inadvertently hid a bug :disappointed: 

![Screenshot from 2023-02-27 19-49-06](https://user-images.githubusercontent.com/37340715/221723294-8fb69c90-8fd2-4f51-85ed-d12690dd3e99.png)

I thought I tested it on the deploy preview and it worked, but clearly I messed something up

Reply:
Follow up on this - #1510

Reply:
Typechecking for the win!